std/uni.d

```diff
@@ -8214,6 +8215,162 @@ unittest //12428
     assert(s == "abcdefghij");
 }
 
+
+// generic toUpper/toLower on whole range, returns range
+private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    static if (!is(Unqual!(ElementEncodingType!Range) : const(dchar)))
+    {
+        import std.utf.byDchar;
+
+        // Decode and try again
+        return toCaser!(indexFn, maxIdx, tableFn, Range)(str.byDchar);
+    }
+    else
+    {
+        static struct ToCaserImpl
+        {
+            this(ref Range r)
+            {
+                this.r = r;
+            }
+
+            @property bool empty()
+            {
+                return !nLeft && r.empty;
+            }
+
+            @property auto front() @safe
```

Comment:
Leave `@safe` to deduction; you don't know whether `front` is safe on the original range type.